go/vt/workflow/manager.go

```diff
@@ -39,9 +39,10 @@ type Factory interface {
 	// variable filled it. This Init method should fill in the
 	// Name and Data attributes, based on the provided args.
 	// This is called during the Manager.Create phase.
-	// TODO(yipeiw): We should extend the interface to pass the topology server
-	// as well. The topology server is needed in the resarding workflow.
-	Init(w *workflowpb.Workflow, args []string) error
+	// For complex workflow (e.g. resharding workflow), Init method should also
```

Comment:
This comment is redundant because the previous sentence already mentions that.

But I agree that it's not very explicit. How about extending this one sentence?

```
This is called during the Manager.Create phase and will initially checkpoint the provided data in the topology. Later calls to Instantiate(), even across process restarts, will re-read the written checkpoint.
```